### ISSUE TITLE
node-persist reading all files to RAM ?

### ISSUE THREAD

**Issue by Vance-ng-vn** [simonlast/node-persist#148]

**Comment by akhoury** [comment]
if you use any of these 4 functions, https://github.com/simonlast/node-persist/blob/master/src/local-storage.js#L133-L166 yes, it seems that is in fact reading everything to ram, temporarily, this should be garbage collected since we don't maintain a ref to `data`. But for most usage, [`getItem`](https://github.com/simonlast/node-persist/blob/master/src/local-storage.js#L212) only reads the target file. Now, granted, if you are writing an insane number of files, that could increase the size of the `q`, can you share how exactly you're using it.

**Comment by Vance-ng-vn** [comment]
> if you use any of these 4 functions, https://github.com/simonlast/node-persist/blob/master/src/local-storage.js#L133-L166 yes, it seems that is in fact reading everything to ram, temporarily, this should be garbage collected since we don't maintain a ref to `data`. But for most usage, [`getItem`](https://github.com/simonlast/node-persist/blob/master/src/local-storage.js#L212) only reads the target file. Now, granted, if you are writing an insane number of files, that could increase the size of the `q`, can you share how exactly you're using it.

I am using it to cache some files in my Node.js app (subtitle files). I need to cache these files for 7 days. However, it seems that node-persist is not suitable for my needs.

Every time it clean up expired files, node-persist loads all of them into RAM. (The total amount of RAM needed remains unchanged compared to caching directly into RAM, even though it's only temporary)

I've switched to using SQLite for temporary storage, and everything is working fine now.

**Comment by akhoury** [comment]
you are right!, cleaning up expired does in fact load everything, that's a bug that i will fix.
